src/xrc-tests/src/tests/misbehavior.rs

```diff
@@ -0,0 +1,387 @@
+use std::collections::HashMap;
+
+use ic_xrc_types::{
+    Asset, AssetClass, ExchangeRate, ExchangeRateMetadata, GetExchangeRateRequest,
+    GetExchangeRateResult,
+};
+use maplit::hashmap;
+
+use crate::{
+    container::{run_scenario, Container},
+    mock_responses, ONE_DAY_SECONDS,
+};
+
+/// This value is derived in the basic_exchange_rates crypto pair portion of the test.
+const CRYPTO_PAIR_BASIC_STD_DEV: u64 = 3_483_761;
+
+/// This value is derived in the basic_exchange_rates crypto fiat pair portion of the test.
+const CRYPTO_FIAT_PAIR_BASIC_STD_DEV: u64 = 80_650_883;
+
+/// This value is derived in the basic_exchange_rates fiat crypto pair portion of the test.
+const FIAT_CRYPTO_PAIR_BASIC_STD_DEV: u64 = 1_474_512;
+
+/// This value is derived from using the common mock dataset (mock_responses::forex::build_common_responses).
+/// A full explanation how on the number is derived can be seen starting in the basic_exchange_rate test on line
+/// 119.
+const FIAT_PAIR_COMMON_DATASET_STD_DEV: u64 = 396_623_626;
+
+/// Setup:
+/// * Deploy mock FOREX data providers and exchanges, some of which are configured to be malicious
+/// * Start replicas and deploy the XRC, configured to use the mock data sources
+///
+/// Runbook:
+/// * Request exchange rate for various cryptocurrency and fiat currency pairs
+/// * Assert that the returned rates correspond to the expected values and that the confidence is lower due to the erroneous responses
+///
+/// Success criteria:
+/// * All queries return the expected values
+///
+/// The expected values are determined as follows:
+///
+/// Crypto-pair (retrieve ICP/BTC rate)
+/// 0. The XRC retrieves the ICP/USDT rate.
+///     a. ICP/USDT rates: [3900000000, 3900000000, 3911000000, 4005000000]
+///         i. There are only 4 rates as the other 3 have been filtered out as they were greater
```

Comment:
This sounds like rates greater than the median rate are always filtered, making the median rate the largest rate among the remaining rates.

This is not quite accurate, right?